docs/ss-migration-guide.md

```diff
@@ -26,6 +26,19 @@ Note that this migration guide describes the items specific to Structured Stream
 Many items of SQL migration can be applied when migrating Structured Streaming to higher versions.
 Please refer [Migration Guide: SQL, Datasets and DataFrame](sql-migration-guide.html).
 
+## Upgrading from Structured Streaming 3.0 to 3.1
+
+- In Spark 3.0 and below, secure Kafka processing needed the following ACLs from driver perspective:
+  * Topic resource describe operation
+  * Topic resource read operation
+  * Group resource read operation
+
+  Since Spark 3.1, offsets are obtained with `AdminClient` instead of `KafkaConsumer` and now the following ACLs needed from driver perspective:
+  * Topic resource describe operation
+
+  Since `AdminClient` in driver is not connecting to consumer group, `group.id` based authorization will not work anymore (executors never done group based authorization).
```

Comment:
I'm not sure about this point. Can you explain more? What the impact it could be?

Reply:
Some of the users authorizing based on `group.id` which will not work with this PR (topic based authorization is a good alternative). Nothing more, nothing less. If you can help how to make it more descriptive it's welcome.

Reply:
What it means `executors never done group based authorization`? Seems on executors, we use `group.id` if it is set by users, no?

Reply:
Many users who I've spoken with thought that driver and executors are doing group based authorization which makes sense from far distance. Both cases `KafkaConsumer` used and both cases `group.id` is set. The truth is that executors using `KafkaConsumer` in a special way:
* `assign` strategy used only
* Auto offset commit is turned off
* Manual offset commit not called

Such case `KafkaConsumer` is not joining to any group and not doing any group based authorization (this is true with and without this PR).

Not relevant but worth to mention, providing the possibility to either prefix or override executor side `group.id` still has value from user perspective so it's not modified.